curvefs/src/client/rpcclient/metaserver_client.cpp

```diff
@@ -497,163 +497,185 @@ bool GroupInodeIdByPartition(
-    uint32_t limit = opt_.batchLimit;
-    // group inodeid by partition
-    std::unordered_map<uint32_t, std::list<uint64_t>> inodeGroups;
-    if (!GroupInodeIdByPartition(fsId, metaCache_, inodeIds, &inodeGroups)) {
+    // group inodeid by partition and batchlimit
+    std::vector<std::list<uint64_t>> inodeGroups;
+    if (!SplitRequestInodes(fsId, inodeIds, &inodeGroups)) {
         return MetaStatusCode::NOT_FOUND;
     }
 
     // send rpc
     for (const auto &it : inodeGroups) {
-        auto iter = it.second.begin();
-        while (iter != it.second.end()) {
-            uint64_t inodeId = *iter;
-            auto task = RPCTask {
-                metaserverClientMetric_->batchGetInodeAttr.qps.count << 1;
-                LatencyUpdater updater(
-                    &metaserverClientMetric_->batchGetInodeAttr.latency);
-                BatchGetInodeAttrRequest request;
-                BatchGetInodeAttrResponse response;
-                request.set_poolid(poolID);
-                request.set_copysetid(copysetID);
-                request.set_partitionid(partitionID);
-                request.set_fsid(fsId);
-                request.set_appliedindex(
-                    metaCache_->GetApplyIndex(CopysetGroupID(poolID,
-                                              copysetID)));
-                uint32_t batchLimit = limit;
-                while (iter != it.second.end() && batchLimit > 0) {
-                    request.add_inodeid(*iter);
-                    iter++;
-                    batchLimit--;
-                }
-                curvefs::metaserver::MetaServerService_Stub stub(channel);
-                stub.BatchGetInodeAttr(cntl, &request, &response, nullptr);
-
-                if (cntl->Failed()) {
-                    metaserverClientMetric_->batchGetInodeAttr.eps.count << 1;
-                    LOG(WARNING) << "BatchGetInodeAttr Failed, errorcode = "
-                                 << cntl->ErrorCode()
-                                 << ", error content:" << cntl->ErrorText()
-                                 << ", log id = " << cntl->log_id();
-                    return -cntl->ErrorCode();
-                }
+        uint64_t inodeId = *it.begin();
+        auto task = RPCTask {
+            metaserverClientMetric_->batchGetInodeAttr.qps.count << 1;
+            LatencyUpdater updater(
+                &metaserverClientMetric_->batchGetInodeAttr.latency);
+            BatchGetInodeAttrRequest request;
+            BatchGetInodeAttrResponse response;
+            request.set_poolid(poolID);
+            request.set_copysetid(copysetID);
+            request.set_partitionid(partitionID);
+            request.set_fsid(fsId);
+            request.set_appliedindex(
+                metaCache_->GetApplyIndex(CopysetGroupID(poolID,
+                                            copysetID)));
+            for (const auto &id : it) {
+                request.add_inodeid(id);
+            }
```

Comment:
might be a shortcut, `*request.mutable_inode() = { it.begin(), it.end() }`

Reply:
fixed